docs_src/design/adr/0026-Common Configuration.md

```diff
@@ -19,9 +19,9 @@ The following flow chart demonstrates the bootstrapping of each services' config
 ![](common-config-images/EdgeX 2.x Configuration bootstrapping flowchart.png)
 
 ## Proposed Design
-The configuration settings that are common to all services will be partitioned out into a separate common configuration source. This  common configuration source will be pushed into the Configuration Provider by a new `core-common-config` service. When the Configuration Provider is not used the `core-common-config` service will provide the common configuration via a RESTful endpoint.
+The configuration settings that are common to all services will be partitioned out into a separate common configuration source. This common configuration source will be pushed into the Configuration Provider by the new `core-common-config-bootstrapper` service.
 
-During bootstrapping, each service will either load the common configuration from the Configuration Provider or via URI to some endpoint that provides the common configuration. This could be the new service's REST endpoint. Each service will have additional private configuration, which may overwrite and/or extend the common configuration.  
+During bootstrapping, each service will either load the common configuration from the Configuration Provider or via URI to some endpoint that provides the common configuration. Each service will have additional private configuration, which may overwrite and/or extend the common configuration.  

```

Comment:
> Each service will have additional private configuration, which may overwrite and/or extend the common configuration. 

I've [suggested the removal](https://github.com/edgexfoundry/edgex-docs/pull/909#discussion_r1042433163) of this statement in my previous review, unfortunately without explanation. If you insist on keeping it:

1. The service will not overwrite/replace the common config. It overrides them ([as explained in UCR](https://docs.edgexfoundry.org/2.4/design/ucr/Common%20Configuration/#with-file-based-configuration-provider)):
```suggestion
During bootstrapping, each service will either load the common configuration from the Configuration Provider or via URI to some endpoint that provides the common configuration. Each service will have additional private configuration, which may override and/or extend the common configuration.  
```

2. Can a service's private configuration "override" the entire common configuration? If yes, this means that we can run a service with neither a CP nor a common config source (as possible with EdgeX 2), making the following mandates obsolete:

> L44: If the `-cp/--configProvider` option is not specified and the `-cc/--commonConfig` option is not specified, then the service will error out.

> L52: 3. If a common configuration is not provided by one of the means listed, then the services cannot start.

Reply:
@farshidtz makes a really good point - a complete override is equivalent to what we have today (score one for backwards compatibility!), and does mean that we don't have to state L44 about erroring out.

Reply:
@farshidtz @TomBrennan-Eaton 
If the private configuration isn't a complete one (as we have it today), then the service will get random errors on start up due to incomplete config. The cleaner thing may be be to require the user to specify -cc/--commonConfigis set to None in order to use a private configuration (as the backwards compatibility option).

Reply:
@ejlee3 I believe this an implementation issue which can be solved by improving the error handing of config loader. The same would happen if a user comments out or removes some of the other private configurations.

It can be as simple as:

> #### Options for Providing the Common Configuration
>
> 1. From the configuration provider, using the `-cp/--configProvider` command line option
> 2. When the Configuration Provider is not used, the `-cc/--commonConfig` command line option ~~or the `EDGEX_COMMON_CONFIG` environment variable~~ may be specified using 
>    1. An HTTP endpoint that returns the common configuration file in response
>    2. A local file that contains the common configuration
> 3. **From the main configuration file, alongside other private configurations**

I've removed the env var part because it is redundant and misleading (`-cp` doesn't have env override and only `-cc` has it?). The common config variable override is already explained above the text.

Reply:
@farshidtz The implementation issue cannot be solved by improving the error handling of the config loader. The config loader has no knowledge of what configuration is required and shouldn't at that level. 

I added the env variable for the `-cp` option since it is a better option. I also re-worded the sections to use the private configuration.